Common/src/main/java/com/tabnineCommon/chat/ChatDisabledComponents.kt

```diff
@@ -1,60 +1,44 @@
+import javax.swing.JEditorPane
+import javax.swing.event.HyperlinkEvent
+import javax.swing.event.HyperlinkListener
+
+val LOGIN_BUTTON = "<p>Received Tabnine Chat beta access? Please make sure to <a href=\"https://app.tabnine.com/login\">sign in</a></p>".trimIndent()
+
+fun createChatDisabledJPane(isLoggedIn: Boolean = false): JEditorPane {
+    val pane = JEditorPane(
+        "text/html",
+        """<html>
                 <body style="padding: 20px;">
                     <div style="font-size: 12px; margin-bottom: 8px"><b>Tabnine Chat is currently in Beta</b></div>
                     <div style="font-size: 10px;">
                         <p style="margin-bottom: 8px">We understand that waiting for this awesome feature isn’t easy, but we guarantee it will be worth it.</p>
-                        <p style="margin-bottom: 8px">Tabnine Chat will soon be available to all users, and we'll make sure to keep you informed. Thank you for your patience! <a href=""> Learn more</a></p>
-                        <p>Received Tabnine Chat beta access? Please ensure you're signed in</p>
+                        <p style="margin-bottom: 8px">Tabnine Chat will soon be available to all users, and we'll make sure to keep you informed. Thank you for your patience! <a href="https://www.tabnine.com/#ChatSection"> Learn more</a></p>
+                        ${if (!isLoggedIn) LOGIN_BUTTON else ""}
                     </div>
                 </body>
             </html>
-""".trimIndent()
-
-fun createChatDisabledJLabel(): JLabel {
-    val label = JLabel(
-        CHAT_DISABLED_PAGE, SwingConstants.CENTER
-    )
-
-    label.border = BorderFactory.createEmptyBorder(10, 0, 0, 0)
-    label.addMouseListener(NaiveOpenLinkListener("https://www.tabnine.com/#ChatSection"))
-
-    return label
-}
-
-class NaiveOpenLinkListener(private val link: String) : MouseListener {
-    override fun mouseClicked(e: MouseEvent?) {
-        try {
-            Desktop.getDesktop().browse(URI(link))
-            return
-        } catch (e1: IOException) {
-            e1.printStackTrace()
-        } catch (e1: URISyntaxException) {
-            e1.printStackTrace()
-        }
-    }
-
-    override fun mousePressed(e: MouseEvent?) {
-    }
-
-    override fun mouseReleased(e: MouseEvent?) {
-    }
-
-    override fun mouseEntered(e: MouseEvent?) {
+        """.trimIndent()
+    ).apply {
+        isEditable = false
+        isOpaque = false
     }
+    pane.addHyperlinkListener(
+        HyperlinkListener { it ->
+            if (it.eventType !== HyperlinkEvent.EventType.ACTIVATED) {
+                return@HyperlinkListener
+            }
+            if (it.url.toString() == "https://www.tabnine.com/#ChatSection") {
+                BrowserUtil.browse(it.url.toString())
+            } else if (it.url.toString() == "https://app.tabnine.com/login") {
+                DependencyContainer.instanceOfBinaryRequestFacade().executeRequest(LoginRequest {})
```

Comment:
is this the same as the login from the status bar?

Reply:
yes!